pkg/ottl/parser.go

```diff
@@ -480,6 +481,35 @@ func (p *Parser[K]) ParseValueExpression(raw string) (*ValueExpression[K], error
 	}
 
 	return &ValueExpression[K]{
-		getter: getter,
+		getter: &StandardGetSetter[K]{
+			Getter: func(ctx context.Context, tCtx K) (any, error) {
+				val, err := getter.Get(ctx, tCtx)
+				if err != nil {
+					return nil, err
+				}
+				switch v := val.(type) {
+				case map[string]any:
```

Comment:
I'm not sure if this condition is still possible, considering the mapGetter now only returns `pcommon.Map`